exporter/factory.go

```diff
@@ -56,9 +64,12 @@ type FactoryV2 interface {
 	// CreateTraceReceiverV2 creates a trace receiver based on this config.
 	// If the receiver type does not support tracing or if the config is not valid
 	// error will be returned instead.
-	CreateTraceExporterV2(logger *zap.Logger, cfg configmodels.Exporter) (TraceExporterV2, error)
+	CreateTraceExporterV2(ctx context.Context, params CreationParams, cfg configmodels.Exporter) (TraceExporterV2, error)
```

Comment:
Sorry, this didn't occur to me until just now. Do we need `V2` in these names like CreateTraceExporterV2? Since the type signatures are different I think that is enough to distinguish `Factory` and `FactoryV2` by keeping `CreateTraceExporter` with just different parameters. From some quick testing this seems work.

If for some reason we want components to be explicit about their version we could have the factory have a `Version()` method but doesn't look like that's needed yet.

Reply:
In that case it will make impossible for a factory to implement both interfaces, which may be desirable during transition period. Let me think a bit.

Side note: I cannot find the relevant part of Golang spec that describes interface method matching and thus making what you propose compliant with the spec - do you know where is it formally defined?

Reply:
I suggest that we merge as is to avoid more delays and I will look into this separately.